internal/core/data/application/reading.go

```diff
@@ -465,3 +471,54 @@ func countBasedEventRetention(deviceName string, autoEvent models.AutoEvent, dic
 	}
 	return nil
 }
+
+func purgeNoneAutoEventsByDevice(device models.Device, dic *di.Container) errors.EdgeX {
+	if len(device.ProfileName) == 0 {
+		return nil
+	}
+	sources, err := noneAutoEventSourcesByDevice(device, dic)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	for _, source := range sources {
+		autoEvent := models.AutoEvent{SourceName: source} // Empty AutoEvent.Retention will apply default retention policy
+		err = purgeEventsByAutoEvent(device.Name, autoEvent, dic)
+		if err != nil {
+			return errors.NewCommonEdgeXWrapper(err)
+		}
+	}
+	return nil
+}
+
+// noneAutoEventSourcesByDevice returns sources that are not defined in the device's auto events
+func noneAutoEventSourcesByDevice(device models.Device, dic *di.Container) ([]string, errors.EdgeX) {
+	profileClient := bootstrapContainer.DeviceProfileClientFrom(dic.Get)
+	profile, err := profileClient.DeviceProfileByName(context.Background(), device.ProfileName)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.Kind(err), fmt.Sprintf("retrieve event sources not in auto events from the device '%s'", device.Name), err)
+	}
+	var sources []string
+	for _, r := range profile.Profile.DeviceResources {
+		if contained := slices.ContainsFunc(device.AutoEvents, func(event models.AutoEvent) bool {
+			return event.SourceName == r.Name
+		}); contained {
+			continue
+		}
+		if r.Properties.ReadWrite == common.ReadWrite_W {
+			continue
+		}
+		sources = append(sources, r.Name)
+	}
+	for _, c := range profile.Profile.DeviceCommands {
+		if contained := slices.ContainsFunc(device.AutoEvents, func(event models.AutoEvent) bool {
+			return event.SourceName == c.Name
+		}); contained {
+			continue
+		}
+		if c.ReadWrite == common.ReadWrite_W {
```

Comment:
Please add comment to explain why skip device command with ReadWrite_W

Reply:
Added the comment.